drivers/i2c/i2c_nrfx_twim.c

```diff
@@ -52,19 +51,12 @@ static int i2c_nrfx_twim_transfer(const struct device *dev,
 		.address = addr
 	};
 
-	/* If for whatever reason the TWIM peripheral is still not initialized
-	 * at this point, try to initialize it now.
-	 */
-	if (!dev_data->twim_initialized && init_twim(dev) < 0) {
-		return -EIO;
-	}
-
 	k_sem_take(&dev_data->transfer_sync, K_FOREVER);
 
 	/* Dummy take on completion_sync sem to be sure that it is empty */
 	k_sem_take(&dev_data->completion_sync, K_NO_WAIT);
 
-	nrfx_twim_enable(&dev_config->twim);
+	(void)pm_device_runtime_get(dev);
```

Comment:
When `nrfx_twim_bus_recover` is called, the peripheral needs to be disabled (so that the pins are released) and in the current form this is not the case.
Actually, couldn't the calls to `nrfx_twim_enable/nrfx_twim_disable` stay here and instead be removed from `twim_nrfx_pm_action` (is that a problem if the shim reduces power consumption even if not instructed to)?